.github/workflows/coverage.yml

```diff
@@ -12,18 +12,88 @@ on:
     steps:
     - uses: actions/checkout@v4
+      with:
+        # Set branch so that can commit to it later
+        ref: ${{ github.head_ref }}
     - name: Set up Python 3.11
       uses: actions/setup-python@v5
       with:
         python-version: "3.11"
         cache: pip
-        cache-dependency-path: requirements-test-3.11.txt
+        cache-dependency-path: requirements-dev-3.11.txt
     - name: Upgrade pip
       run: python -m pip install --upgrade pip
     - name: Install test dependencies
-      run: pip install --no-dependencies -r requirements-test-3.11.txt
+      run: pip install --no-dependencies -r requirements-dev-3.11.txt
     - name: Assess coverage of unit tests
       run: pytest tests/unit --cov
+    - name: Extract total coverage percentage
+      id: cov
+      # Need to run this in own step to ensure coverage is evaluated and cast to string
+      run: echo "percentage=$( coverage report --format=total )" >> $GITHUB_OUTPUT
+    - name: Choose badge colour
+      id: design
+      env:
+        percentage: ${{ steps.cov.outputs.percentage }}
+      run: |
+        echo "colour=${{
+          env.percentage >= 90 && 'green' ||
+          env.percentage >= 70 && 'yellow' ||
+          env.percentage >= 50 && 'orange' ||
+          'red'
+        }}" >> $GITHUB_OUTPUT
+    - name: Create badges directory if necessary
+      run: mkdir -p .github/badges
+    - name: Store copy of old badge config for comparison
+      id: old_file
+      run: |
+        if [ -f $badge_path ]; then
+          var=$( cat $badge_path )
+        else
+          var=""
+        fi
+        {
+          echo "file<<EOF"
+          echo $var
+          echo EOF
+        } >> $GITHUB_OUTPUT
+    - name: Generate badge config YAML
+      env:
+        p: ${{ steps.cov.outputs.percentage }}
+        c: ${{ steps.design.outputs.colour }}
+      run: |
+        echo "coverage = ${{ env.p }}% with colour ${{ env.c }}"
+        {
+          echo "---"
+          echo "badge: 'Coverage-${{ env.p }}%-${{ env.c }}'"
+        } > $badge_path
+    - name: Store copy of new badge config for comparison
+      id: new_file
+      run: |
+        {
+          echo "file<<EOF"
+          echo $( cat $badge_path )
+          echo EOF
+        } >> $GITHUB_OUTPUT
+    - name: Commit badge config if changed
+      if: steps.old_file.outputs.file != steps.new_file.outputs.file
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        destination_branch: ${{ github.head_ref }}
+        new_badge: ${{ steps.new_file.outputs.file }}
+      run: |
+        export message="chore: update coverage badge"
+        export sha=$( git rev-parse $destination_branch:$badge_path )
+        export content=$( base64 -i $badge_path )
+        gh api --method PUT /repos/:owner/:repo/contents/$badge_path \
+          --field message="$message" \
+          --field content="$content" \
+          --field branch="$destination_branch" \
+          --field sha="$sha"
```

Comment:
Is this possible via the [checkout](https://github.com/actions/checkout) action?

Reply:
I tried using `git push` relying on `checkout`. However, the commits aren't signed and would require "signing" the CLA on behalf of whatever email address we give to the bot. Using gh API is the way to get them signed properly.